room/runtime/src/main/java/androidx/room/SQLiteCopyOpenHelper.java

```diff
@@ -101,6 +120,21 @@ public synchronized SupportSQLiteDatabase getReadableDatabase() {
         if (!mVerified) {
             verifyDatabaseFile();
             mVerified = true;
+            if(mDatabaseConfiguration.prepackagedCallbacks != null) {
+                // Temporarily open database using FrameworkSQLiteOpenHelper
+                SupportSQLiteOpenHelper helper = createFrameworkOpenHelper();
+                if (helper != null) {
+                    SupportSQLiteDatabase db = helper.getReadableDatabase();
+                    // Invoke callbacks passing db as a param
+                    callOnOpenPrepackagedDatabaseCallbacks(db);
```

Comment:
`callOnOpenPrepackagedDatabaseCallbacks` should be inside a try-finally where in the finally block we close the DB.

Reply:
I made the changes but I'm still not sure if it's ok. There are a couple of things that may throw:
1. Reading the db version
2. Calling `getReadable/WritableDatabase()`
3. Calling `database.close()`

Take a look please and let me know if you'd like to change anything